include/simdjson/dom/document_stream.h

```diff
@@ -204,6 +215,9 @@ class document_stream {
 #endif // SIMDJSON_THREADS_ENABLED
 
   friend class dom::parser;
+
+  size_t doc_index{};
```

Comment:
It is somewhat unfortunate that we need to track the index while most users will never access it. However, it seems that there is no safe way to recover it (return in time) after the document has been parsed. On the *plus* side, this is probably an important feature. You do what to know where, in the damn large file, the parse is at... and how many bytes. And if the parsing fails, you do want to know *where* it failed. So this could be a handy debugging tool.